Dockerfile

```diff
@@ -0,0 +1,33 @@
+#Another attempt at dockerizing frontend
+
+
+# Stage 1: Compile and Build angular codebase
+
+# Use official node image as the base image
+FROM node:latest as build
+
+# Set the working directory
+WORKDIR /app
+
+# Add the source code to app
+COPY ./p3-frontend .
+
+# Install all the dependencies
+#RUN cd ./p3-frontend
+RUN npm install
+
+# Generate the build of the application
+RUN npm run build
+
+
+# Stage 2: Serve app with nginx server
+
+# Use official nginx image as the base image
+FROM nginx:latest
+
+# # Copy the build output to replace the default nginx contents.
+COPY --from=build /app/dist/p3-frontend /usr/share/nginx/html
```

Comment:
that's not the only needed thing to host an angular app on nginx. [you also need to configure nginx](https://v11.angular.io/guide/deployment#routed-apps-must-fallback-to-indexhtml) so that all 404 errors that are probably attempts to go to some route inside the app are redirected to index.html. once the browser loads index.html, all the javascript referenced from it looks at the URL and renders the right stuff. this is how "single-page apps" work.

the angular docs don't explain 100% of how to accomplish this, only about half. if you want i can walk you through it, just let me know.

this isn't something i'd expect you to know about at all. if the devs were around, it's be up to them to tell you about this.

Reply:
removed docker files from this feature branch.

Reply:
I think I am ready to learn what you mean from this step now if you have the time tomorrow.  Thank you!

Reply:
Okay, I think I did this like you showed us.